### PR TITLE
JC-1723 Optimistic lock handlers replaced with retry template

### PR DESCRIPTION
**Pull request by mixas27**
Custom optimistic lock handlers replaced with spring retry mechanism
Retry template set up to retry methods 3 times after HibernateOptimisticLockingFailureException

### CODE REVIEW

File: jcommune-view/jcommune-web-controller/src/main/java/org/jtalks/jcommune/web/controller/CodeReviewController.java [hunk 1]
```diff
@@ -115,11 +115,11 @@ public ModelAndView showNewCodeReviewPage(@RequestParam(BRANCH_ID) Long branchId
      * @throws NotFoundException when branch not found
      */
     @RequestMapping(value = "/reviews/new", method = RequestMethod.POST)
-    public ModelAndView createCodeReview(@Valid @ModelAttribute TopicDto topicDto,
+    public ModelAndView createCodeReview(@Valid @ModelAttribute final TopicDto topicDto,
                                          BindingResult result,
                                          @RequestParam(BRANCH_ID) Long branchId) throws NotFoundException {
         Branch branch = branchService.get(branchId);
-        Topic topic = topicDto.getTopic();
+        final Topic topic = topicDto.getTopic();
```
Comment (ctapobep) [inline jcommune-view/jcommune-web-controller/src/main/java/org/jtalks/jcommune/web/controller/CodeReviewController.java]: Why do you need `final` here?

File: jcommune-view/jcommune-web-controller/src/main/java/org/jtalks/jcommune/web/controller/CodeReviewController.java [hunk 2]
```diff
@@ -131,25 +131,14 @@ public ModelAndView createCodeReview(@Valid @ModelAttribute TopicDto topicDto,
...
 
-        Topic createdTopic = topicModificationService.createTopic(topic, topicDto.getBodyText());
-
-        return new ModelAndView(REDIRECT_URL + createdTopic.getId());
-    }
-
-    private Topic createCodeReviewWithLockHandling(Topic topic, TopicDto topicDto) throws NotFoundException {
-        for (int i = 0; i < UserController.LOGIN_TRIES_AFTER_LOCK; i++) {
-            try {
+        Topic createdTopic = retryTemplate.execute(new RetryCallback<Topic, NotFoundException>() {
```
Comment (ctapobep) [inline jcommune-view/jcommune-web-controller/src/main/java/org/jtalks/jcommune/web/controller/CodeReviewController.java]: Interesting - shouldn't it accept the number of times it's going to retry?

File: jcommune-view/jcommune-web-controller/src/main/java/org/jtalks/jcommune/web/controller/PostController.java [hunk 3]
```diff
@@ -124,32 +130,19 @@ public PostController(PostService postService, BreadcrumbBuilder breadcrumbBuild
      * @throws NotFoundException when post was not found
      */
     @RequestMapping(method = RequestMethod.DELETE, value = "/posts/{postId}")
-    public ModelAndView delete(@PathVariable(POST_ID) Long postId)
+    public ModelAndView delete(@PathVariable(POST_ID) final Long postId)
```
Comment (ctapobep) [inline jcommune-view/jcommune-web-controller/src/main/java/org/jtalks/jcommune/web/controller/PostController.java]: This `final` is also unnecessary.

File: jcommune-view/jcommune-web-controller/src/main/java/org/jtalks/jcommune/web/controller/CodeReviewController.java (hunk 1 above)
Comment (mixas27) [inline jcommune-view/jcommune-web-controller/src/main/java/org/jtalks/jcommune/web/controller/CodeReviewController.java]: Methods of anonymous classes requires `final`

File: jcommune-view/jcommune-web-controller/src/main/java/org/jtalks/jcommune/web/controller/UserController.java [hunk 4]
```diff
@@ -529,4 +512,26 @@ public JsonResponse sendEmailConfirmation(@RequestParam("id")long id){
         }
         return new JsonResponse(JsonResponseStatus.SUCCESS);
     }
+
+    private class LoginRetryCallback implements RetryCallback<AuthenticationStatus, Exception> {
+
+        private LoginUserDto loginUserDto;
```
Comment (ctapobep) [inline jcommune-view/jcommune-web-controller/src/main/java/org/jtalks/jcommune/web/controller/UserController.java]: While here `final` is useful ;)

File: jcommune-view/jcommune-web-controller/src/main/java/org/jtalks/jcommune/web/controller/CodeReviewController.java (hunk 2 above)
Comment (mixas27) [inline jcommune-view/jcommune-web-controller/src/main/java/org/jtalks/jcommune/web/controller/CodeReviewController.java]: It configurated in applicationContext-controller.xml

**Comment by ctapobep** [comment]
Merged

**Pull request closed without merging**